docs/how-to-migrate-from-smartlock.md

```diff
@@ -0,0 +1,791 @@
+explicitly declare the ID token providers they support (e.g. Google,
+Microsoft), along with any provider-specific parameters that may be required.
+Currently, OpenYOLO directly supports specifying your application’s OpenID
+Connect client ID and an ID token nonce value.
+
+### Receiving a hint response
+
+Hint responses are delivered to the application via onActivityResult in both
+the Smart Lock and OpenYOLO APIs. Both indicate the success or failure of the
+request via a result code and Intent extras, though the data is structurally
+different.
+
+With Smart Lock for Passwords:
+
+```java
+@Override protected void onActivityResult(
+    int requestCode, int resultCode, Intent data) {
+
+  if (requestCode == RC_HINT && resultCode == RESULT_OK) {
+    Credential credential =
+        data.getParcelableExtra(Credential.EXTRA_KEY);
+    if (credential == null) {
+      authenticateManually();
+    } else {
+      processSmartLockCredentialAsHint(credential);
+    }
+  }
+}
+```
+
+With OpenYOLO:
+
+```java
+@Override
+protected void onActivityResult(
+    int requestCode, int resultCode, Intent data) {
+
+  if (requestCode == RC_HINT) {
+    HintRetrieveResult result =
+        mCredentialClient.getHintRetrieveResult(data);
+
+    if (result.getResultCode() !=
+        HintRetrieveResult.CODE_HINT_SELECTED) {
+      authenticateManually();
+      return;
+    }
+
+    Hint hint = result.getHint();
+    processOpenYoloHint(hint);
+  }
+}
+```
+
+With OpenYOLO, the `resultCode` value will always be equal to the value returned
+by `result.getResultCode()`, so either can be used. The standard result codes
+are  all defined as constants on the `HintRetrieveResult` class. The result
+codes in OpenYOLO are more detailed than those returned by Smart Lock for
+Passwords, providing more information particularly in the cases where a hint is
+not returned:
+
+|----------------------------------|-------------------------------------------|
+| *OpenYOLO hint result code*      | *Description*                             |
+|----------------------------------|-------------------------------------------|
+| `CODE_HINT_SELECTED`             | The user selected a hint, which can be    |
+|                                  | retrieved by calling the `getHint()`      |
+|                                  | method.                                   |
+|----------------------------------|-------------------------------------------|
+| `CODE_BAD_REQUEST`               | The request was malformed in some way.    |
+|                                  | `getHint()` method will return null.      |
+|----------------------------------|-------------------------------------------|
+| `CODE_NO_HINTS_AVAILABLE`        | There are no hints available in the       |
+|                                  | user’s credential manager that match the  |
+|                                  | request type. The getHint method will     |
```

Comment:
Nit: `` `getHint()` `` is used in the two rows above, whereas `getHint` is used here.